templates/default.yaml

```diff
@@ -420,8 +420,10 @@ networks:
 # # default: guestPortRange: [1, 65535]
 # # default: hostPortRange: [1, 65535]
 #
-# - guestPort: 8888
-#   ignore: true (don't forward this port)
+# - guestIP: 0.0.0.0 # otherwise defaults to 127.0.0.1
+# # default: guestPortRange: [1, 65535]
+#   proto: any       # tcp and udp
+#   ignore: true     # don't forward this port
```

Comment:
This format is very confusing, mixing incline comments and comment lines. It will be much more clear like this:

```yaml
# Disable port forwarding (any protocol, any guest IP):
# - ignore: true
#   proto: any
#   guestIP: "0.0.0.0"
```

Reply:
> This format is very confusing, mixing incline comments and comment lines.

It is supposed to be valid YAML if you just uncomment the lines by deleting the first 2 characters. So the first line would still need a double comment marker:

```yaml
# # Disable port forwarding (any protocol, any guest IP):
# - ignore: true
#   proto: any
#   guestIP: "0.0.0.0"
```

We just don't have a comment on top for any of the other examples in that section, so I think it is still confusing, but I'm fine changing it, if you think it is better. Should probably add explanations to all the example rules then.

Ultimately I think we should have a page in the docs that documents all the individual settings, and then shows some examples in a cookbook style. The only problem is that it then becomes even more likely that documentation won't get kept in sync with the implementation.

Reply:
> > This format is very confusing, mixing incline comments and comment lines.
> 
> It is supposed to be valid YAML if you just uncomment the lines by deleting the first 2 characters. So the first line would still need a double comment marker:
> 
> ```yaml
> # # Disable port forwarding (any protocol, any guest IP):
> # - ignore: true
> #   proto: any
> #   guestIP: "0.0.0.0"
> ```

It is valid if you use only single comment for the comment line, and it looks better. 

> We just don't have a comment on top for any of the other examples in that section, so I think it is still confusing, but I'm fine changing it, if you think it is better. Should probably add explanations to all the example rules then.

True, this change should be done for the entire file. We can keep the current format for now to keep the file more consistent.

> Ultimately I think we should have a page in the docs that documents all the individual settings, and then shows some examples in a cookbook style. 

Yes!

> The only problem is that it then becomes even more likely that documentation won't get kept in sync with the implementation.

If the documentation is part of the source I don't think this is an issue. You have to remember to update templates/default.yaml or docs/cookbook.md, what is the difference?

Reply:
> If the documentation is part of the source I don't think this is an issue. You have to remember to update templates/default.yaml or docs/cookbook.md, what is the difference?

Shouldn't be, but I just noticed #2791 recently, so obviously we do have a problem updating the docs. We just need to all train ourselves to look for it in code reviews.

Reply:
> It is valid if you use only single comment for the comment line, and it looks better.

The first line is not valid YAML if you uncomment the whole block (remove the first two columns):

![CleanShot 2024-11-14 at 13 49 57@2x](https://github.com/user-attachments/assets/477484ae-c4e8-422b-a245-19e18f7f4a90)

This is something you can do with many IDEs by marking the block and pressing e.g `⌘-/` (JetBrains) or `⌘-;` (Emacs).

Anyways, it isn't super important, and I already agreed that the current format is not easy to understand unless you already know what it means.

Reply:
I've removed the `guestPortRange` double-comment and folded it into the comment on `ignore`.

I'm not willing to put the top comment in that breaks the current schema unless we rework all of the examples.